docs/cmd/kn_subscription_create.md

```diff
@@ -0,0 +1,46 @@
+## kn subscription create
+
+Create a subscription
+
+### Synopsis
+
+Create a subscription
+
+```
+kn subscription create NAME
+```
+
+### Examples
+
+```
+
+  # Create a subscription 'sub0' from InMemoryChannel 'pipe0' to a subscriber ksvc 'receiver'
+  kn subscription create sub0 --channel imcv1beta1:pipe0 --subscriber ksvc:receiver
+
+  # Create a subscription 'sub1' from KafkaChannel 'k1' to ksvc 'mirror', reply to a broker 'nest' and DeadLetterSink to a ksvc 'bucket'
+  kn subscription create sub1 --channel messaging.knative.dev:v1alpha1:KafkaChannel:k1 --subscriber mirror --reply broker:nest --dead-letter-sink bucket
+```
+
+### Options
+
+```
+      --channel string            Specify the channel to subscribe to, in the format '--channel Group:Version:Kind:Name'. You can use channel type aliases from kn config with this flag. You can also refer inbuilt channel type InMemoryChannel using alias 'imc' like '--type imc:CHANNEL_NAME'. Examples: '--channel messaging.knative.dev:v1alpha1:KafkaChannel:k1' for specifying explicit Group:Version:Kind:Name.
+      --dead-letter-sink string   Addressable sink for events. You can specify a broker, Knative service or URI. Examples: '--dead-letter-sink broker:nest' for a broker 'nest', '--dead-letter-sink https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--dead-letter-sink ksvc:receiver' or simply '--dead-letter-sink receiver' for a Knative service 'receiver'. If prefix is not provided, it is considered as a Knative service.
+  -h, --help                      help for create
+  -n, --namespace string          Specify the namespace to operate in.
+      --reply string              Addressable sink for events. You can specify a broker, Knative service or URI. Examples: '--reply broker:nest' for a broker 'nest', '--reply https://event.receiver.uri' for an URI with an 'http://' or 'https://' schema, '--reply ksvc:receiver' or simply '--reply receiver' for a Knative service 'receiver'. If prefix is not provided, it is considered as a Knative service.
```

Comment:
As @grantr pointed out, the original difference was that a _sink_ don't return a CE but a _subscriber_ does. According to https://github.com/knative/eventing/blob/master/docs/spec/data-plane.md , this definition does not really shine through (i.e. a sink _can_ also return a value). Also `kn trigger` uses `--sink` but fills in a `subscriber` field internally.

So I tend to the following solution:

* Use `--sink` also for subscriptions
* Use `--reply-sink` instead of only `--reply` to make is consistent with `--dead-letter-sink`

Reply:
After some discussion in https://knative.slack.com/archives/C9JP909F0/p1601309985065600 I like that suggestion best:

* Use `--sink` also for subscriptions
* Use  `--sink-reply` and `--sink-dead-letter` to group all sink typed options together

Reply:
I put the discussion about the proper option names to today's wg call.

Reply:
I am fine changing the flag names to better align with other command's flags, should this also go to eventing to change these  fields name in the SPEC to better align ? 

> Use --sink-reply and --sink-dead-letter to group all sink typed options together

I wonder if `sink` should be a suffix than prefix here, this way we dont deviate from the fields in spec (only 5 flags here, so I dont think grouping would be useful enough).

Reply:
well, for `Reply` we would deviate anyway.  So having those together in the list might make sense ?